### PR TITLE
fix(ds): use integer for tag values

### PR DESCRIPTION
**Pull request by thalesmg**
Follow up to https://github.com/emqx/emqx/pull/11906#discussion_r1389115973

### <samp>🤖 Generated by Copilot at e597b2a</samp>

Refactor durable storage modules to use integers as tags for stream and it. This improves performance and memory efficiency for replication and logical timestamping.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Review by savonarola — DISMISSED** [review]

**Review by ieQu1 — DISMISSED** [review]

File: apps/emqx_durable_storage/src/emqx_ds_replication_layer.erl [hunk 1]
```diff
@@ -47,8 +47,8 @@
 %% records over the wire.
 
 %% tags:
--define(stream, stream).
--define(it, it).
+-define(stream, 1).
+-define(it, 2).
```
Comment (keynslug) [inline apps/emqx_durable_storage/src/emqx_ds_replication_layer.erl:51]: Tiny nit. What do you think making them UPPERCASE to disambiguate them from keys?

**Review by keynslug — DISMISSED** [review]
🔥

File: apps/emqx_durable_storage/src/emqx_ds_replication_layer.erl (hunk 1 above)
Comment (thalesmg) [reply apps/emqx_durable_storage/src/emqx_ds_replication_layer.erl]: :heavy_check_mark:

**Review by keynslug — APPROVED** [review]

**Review by savonarola — APPROVED** [review]

**Comment by thalesmg** [comment]
Had to rebase due to conflicts.

**Review by keynslug — APPROVED** [review]

**Pull request merged**